OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,245 @@
+        self.bars[emotion]["count"] += 1
+
+    def _relative_size_estimation(self, visualization_size, bar_amount: int = 5):
+        bar_section_size = visualization_size / bar_amount
+        bar_size = int(bar_section_size - 2 * bar_section_size * self.emotion_bar_offset_percent)
+        padding = int(bar_section_size - bar_size)
+        emotion_size = (int(bar_size / 3), int(bar_size / 3))
+        return bar_size, padding,  emotion_size
+
+    def draw_bars(self, vis_result):
+        y_offset = int(vis_result.shape[1] * self.y_offset_percent)
+        alpha_mask_overlay = self.overlay[:, :, 3] / 255.0
+        overlay_background = self.overlay[..., :3]
+        vis_result = overlay_image(vis_result, overlay_background, 0, 0, alpha_mask_overlay)
+        start_x, start_y = np.clip(vis_result.shape[1] - self.bar_size - int(self.padding / 2), 0, vis_result.shape[1]), \
+                           np.clip(vis_result.shape[0] - self.bar_size - y_offset, 0,
+                                   vis_result.shape[0])
+        for emotion, emotion_bar in self.bars.items():
+            percent = self.get_current_emotion_percent(emotion)
+            text = f'{emotion.upper()} - {percent}%'
+            vis_result = add_class_names_and_percents(vis_result, [start_x, int(vis_result.shape[0] - y_offset / 2)],
+                                                      text)
+            alpha_mask = emotion_bar["bar"].alpha_mask
+            vis_result = overlay_image(vis_result, emotion_bar["bar"].bar, start_x, start_y, alpha_mask)
+            start_x = np.clip(start_x - self.bar_size - self.padding, 0, vis_result.shape[1])
+        return vis_result
+
+    def get_total_update_counts(self):
+        return sum([self.bars[x]["count"] for x in self.bars.keys()])
+
+    def update_bars(self, top_emotion):
+        self._increment_count(top_emotion)
+        total_amount = self.get_total_update_counts()
+        # update all others bars
+        for emotion in self.bars.keys():
+            progress = self.bars[emotion]["count"] / total_amount
+            self.bars[emotion]["bar"].draw(progress)
+
+    def save_statistic_to_json(self):
+        with open(self.output_statistics_path, 'w') as outfile:
+            statistics = {x: y["count"] for x, y in self.bars.items()}
+            json.dump(statistics, outfile, indent=1, sort_keys=True)
+
+    def get_current_emotion_percent(self, emotion):
+        if self.get_total_update_counts() != 0:
+            return int((self.bars[emotion]["count"] / self.get_total_update_counts()) * 100)
+        else:
+            return 0
+
+    def create_statistics_pie_chart(self, save: bool = False, save_path: str = 'result_statistics_pie_chart.png'):
+        statistics = {x: y["count"] for x, y in self.bars.items()}
+        emotions = list(statistics.keys())
+        emotions_amount = list(statistics.values())
+        total_emotion_amount = sum(emotions_amount)
+        if total_emotion_amount:
```

Comment:
!= 0?

Reply:
it will work fine if total_emotion_amount not 0. 

![image](https://user-images.githubusercontent.com/36930868/111913368-e3f61600-8a7e-11eb-8d76-59b99d90e1e6.png)

Reply:
It definitely will but we should do it for better readability :)

We use type hints for the same purpose